MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClientDefault.java

```diff
@@ -456,23 +456,23 @@ public int getProductPageSize() {
     }
 
     @Override
-    public CancellableTask getProductPage(int page, Callback<List<Product>> callback) {
-        return productService.getProductPage(page, callback);
+    public CancellableTask getProducts(int page, Callback<List<Product>> callback) {
+        return productService.getProducts(page, callback);
     }
 
     @Override
-    public Observable<List<Product>> getProductPage(int page) {
-        return productService.getProductPage(page);
+    public Observable<List<Product>> getProducts(int page) {
+        return productService.getProducts(page);
     }
 
     @Override
-    public CancellableTask getProductWithHandle(String handle, Callback<Product> callback) {
-        return productService.getProductWithHandle(handle, callback);
+    public CancellableTask getProductByHandle(String handle, Callback<Product> callback) {
```

Comment:
Sure, that makes sense

On Tue, Jun 14, 2016, 10:35 AM Ivan Savytskyi <notifications@github.com>
wrote:

> In
> MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClientDefault.java
> <https://github.com/Shopify/mobile-buy-sdk-android/pull/181#discussion_r66982618>
> :
>
> >      }
> >
> >      @Override
> > -    public CancellableTask getProductWithHandle(String handle, Callback<Product> callback) {
> > -        return productService.getProductWithHandle(handle, callback);
> > +    public CancellableTask getProductByHandle(String handle, Callback<Product> callback) {
>
> Well the issue I see with that is we have getProduct(Long productId) so
> the signature almost the same (different type of parameter only), and could
> confuse user what method is for
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Shopify/mobile-buy-sdk-android/pull/181/files/c2d1019e7b2c831edb3118d7b2895ceced97011e#r66982618>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABqEwJ9HAi3AZfUrp-ySYM5vjZyLibSwks5qLrw1gaJpZM4I1XBX>
> .
>